src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -614,38 +674,48 @@ class WelcomePage extends Disposable {
 	}
 
 	private addExtensionPack(container: HTMLElement, anchorSelector: string) {
-		const btnContainer = container.querySelector(anchorSelector);
-		if (btnContainer) {
-			extensionPacks.forEach((extension, i) => {
-				const a = document.createElement('a');
-				const classes = ['btn', 'btn_secondary', 'a_self_end', 'flex', 'flex_a_center', 'flex_j_center'];
-				const btn = document.createElement('button');
-				const description = document.querySelector('.extension_pack_body');
-				const header = document.querySelector('.extension_pack_header');
+		const btnContainer = container.querySelector(anchorSelector) as HTMLElement;
 
-				a.classList.add(...classes);
-				a.innerText = localize('welcomePage.install', "Install");
-				a.title = extension.title || (extension.isKeymap ? localize('welcomePage.installKeymap', "Install {0} keymap", extension.name) : localize('welcomePage.installExtensionPack', "Install additional support for {0}", extension.name));
-				a.classList.add('installExtension');
-				a.setAttribute('data-extension', extension.id);
-				a.href = 'javascript:void(0)';
-				a.addEventListener('click', e => {
+		if (btnContainer) {
+			extensionPacks.forEach((extension) => {
+
+				const installText = localize('welcomePage.install', "Install");
+				let dropdownBtn = new Button(btnContainer);
```

Comment:
Make sure to register buttons to the disposable store (this._register)

Reply:
done

Reply:
@Charles-Gagnon is adding button to the disposable store enough? should we also add the event handler, right?

Reply:
Well, for buttons we should be using the onDidClick event https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/browser/ui/button/button.ts#L47 which is registered with the button.

@v-bbrady You should be using that instead of setting the href and other stuff directly. 

And for this whole dropdown thing - Hale added a DropdownMenuActionViewItem recently. You should look at reusing that instead of reimplementing (yet another) dropdown menu.

Reply:
You also missed registering this button on line 681